### ISSUE TITLE
Implement the Repository pattern

### ISSUE THREAD

**Issue by BogdanToader25** [gabrielatoader/WPFApp_LibraryManager#16]
Note: Do this after the dependency injection refactoring for services!

Similar to how the services work, you will have to implement repositories that will consist of the methods needed to get, insert, update, delete and other operations you may require to do for a specific table from the database.

These classes will be called from the service classes in a similar way the service classes are called from the UserControls classes. They should also be injected using Dependency Injection so you need to have a similar implementation (have an interface and a class that implements that interface in order to do the DI properly)

Also, these repository classes will replace the existing DbContext static class. Each repository will have a private field
``` private readonly SqlConnection _sqlConnection;``` 
that will be instantiated in the constructor using the connection string:
```
public AuthorRepository()
{
    _sqlConnection = new SqlConnection(ConfigurationManager.ConnectionStrings["WPFApp_LibraryManager.Properties.Settings.LibraryManagerDBConnectionString"].ConnectionString);
}
```

As an example, the AuthorRepository will currently contain only one method: 
``` public List<Author> GetAuthors()``` 
which will query the database, retrieve the list of authors in a DataTable then map it to a list of authors and return that list  (basically, almost everything the method in the service class does now).

This will mean that the method in the service class will change into something like:
```
   public List<Author> GetAuthors()
   {
        List<Author> authorList = new List<Author>();
        
        Author header = new Author();   
        header.Id = 0;
        header.FullName = "- AUTHOR -";
        authorList.Add(header);
        
        var authors = _authorRepository.GetAuthors();
        
        authorList.AddRange(authors);
        
        return authorList;
    }
```

To summarize
The repository classes should:
-  be the only ones that access the database  (or any other data storage integration - API, files on disk, etc)
- map the results to the respective model (or collection of models)

The service class should:
- call the repository class to retrieve/save data from/to the database (or any other data storage integration - API, files on disk, etc)
- do any extra processing is required before the data is displayed/sent to be saved
